logger/logfile.go

```diff
@@ -104,8 +104,14 @@ func (l *LogFile) pruneFiles() error {
 	if err != nil {
 		return err
 	}
-	// Prune if there are more files stored than the configured max
-	stale := len(matches) - l.MaxFiles
+	var stale int
+	if l.MaxFiles == -1 {
```

Comment:
I'm wondering now whether we should just make this check `if l.MaxFiles <= -1`, so that we handle the edge cases of setting this to `-2`, `-10000`. etc. Not really expecting anyone to seriously do that, but better safe than sorry perhaps.

Reply:
Agreed!